admin/cleanup.py

```diff
@@ -0,0 +1,379 @@
+        """
+        for volume in volumes:
+            try:
+                volume.destroy()
+            except:
+                err(None, "Destroying volume.")
+
+
+def _dumps(obj):
+    """
+    JSON encode an object using some visually pleasing formatting.
+    """
+    return json.dumps(obj, sort_keys=True, indent=4, separators=(',', ': '))
+
+
+def _format_time(when):
+    """
+    Format a time to ISO8601 format.  Or if there is no time, just return
+    ``None``.
+    """
+    if when is None:
+        return None
+    return datetime.isoformat(when)
+
+
+def _get_tag(volume, tag_name):
+    """
+    Get a "tag" from an EBS or Rackspace volume.
+
+    libcloud doesn't represent tags uniformly across drivers.  Thus this method
+    only works on drivers specifically account for.
+
+    :raise: ``KeyError`` if the tag is not present.
+    """
+    return volume.extra.get("tags", volume.extra.get("metadata"))[tag_name]
+
+
+@attributes(["destroy", "keep"])
+class VolumeActions(object):
+    """
+    Represent something to be done to some volumes.
+
+    :ivar destroy: Volumes to destroy.
+    :ivar keep: Volumes to keep.
+    """
+
+
+def _existing_file_path_option(option_name, option_value):
+    file_path = FilePath(option_value)
+    if not file_path.exists():
+        raise UsageError(
+            u"Problem with --{}. File does not exist: '{}'.".format(
+                option_name, file_path.path
+            )
+        )
+    return file_path
+
+
+def _yaml_configuration_path_option(option_name, option_value):
+    yaml_path = _existing_file_path_option(option_name, option_value)
+    try:
+        configuration = yaml.safe_load(yaml_path.open())
+    except ParserError as e:
+        raise UsageError(
+            u"Problem with --{}. "
+            u"Unable to parse YAML from {}. "
+            u"Error message: {}.".format(
+                option_name, yaml_path.path, unicode(e)
+            )
+        )
+    return configuration
+
+
+class CleanupCloudResourcesOptions(Options):
+    """
+    """
+    optFlags = [
+        ["dry-run", None,
+         "Just print the calculated actions. Don't delete anything."]
+    ]
+
+    optParameters = [
+        [u"config-file", None, None,
+         u"The config file containing cloud credentials.\n",
+         lambda option_value: _yaml_configuration_path_option(
+             u"config-file", option_value
+         )],
+        [u"volume-lag", None, timedelta(minutes=30),
+         u"The oldest in minutes a volume may be "
+         u"without being considered for deletion.\n",
+         lambda option_value: timedelta(minutes=int(option_value))],
+        [u"marker", None, MARKER,
+         u"The marker which is used "
+         u"as the ``node`` in the acceptance test cluster UUID.",
+         lambda option_value: int(option_value, base=16)]
+    ]
+
+    def postOptions(self):
+        self["dry-run"] = bool(self["dry-run"])
```

Comment:
``t.p.options.Options`` sets the value to 0 or 1. I just wanted to get the types correct as soon as possible.